src/urn.cc

```diff
@@ -262,55 +252,34 @@ urnHandleReply(void *data, StoreIOBuffer result)
         return;
     }
 
-    /* Update reqofs to point to where in the buffer we'd be */
-    urnState->reqofs += result.length;
-
-    /* Handle reqofs being bigger than normal */
-    if (urnState->reqofs >= URN_REQBUF_SZ) {
+    if (!Less(result.offset + result.length, urnState->storeReadBuffer.size())) {
```

Comment:
I kept the old behavior here - however we could process the already received data (of the buffer size) and send it to the client. With the old behavior - client just hangs without receiving anything.